File: euler.py
#!/usr/bin/env python

# Project Euler in Python (2.5)
# John Evans <john@jpevans.com>

import sys

def euler1():
    """Euler #1
    Answer: 233168

    If we list all the natural numbers below 10 that are multiples of 3 or 5,
    we get 3, 5, 6 and 9. The sum of these multiples is 23.

    Find the sum of all the multiples of 3 or 5 below 1000.

    """

    return sum(i for i in range(1, 1000) if i % 3 == 0 or i % 5 == 0)


EULERS = [
    euler1
]

if __name__ == "__main__":
    if len(sys.argv) > 1:
        for n in sys.argv[1:]:
            n = int(n)
            print "%d: %d" % (n, eval("euler%d()" % n))
    else:
        for index, euler in enumerate(EULERS):
            print "%d: %d" % (index+1, euler())

File: euler.php
#!/usr/bin/env php
<?php
// If we list all the natural numbers below 10 that are multiples of 3 or 5,
// we get 3, 5, 6 and 9. The sum of these multiples is 23.
//
// Find the sum of all the multiples of 3 or 5 below 1000.
function euler1() {
   return array_sum(array_filter(range(3, 999),
                    create_function('$x',
                                    'return ($x % 3 == 0) || ($x % 5 == 0);')));
}

$eulers = array(
    1 => euler1,
);

if (isset($argc)) {
    // Main
    if ($argc == 1) {
        // No command line args passed, so run them all.
        foreach ($eulers as $num => $euler) {
            echo "#" . $num . ". " . $euler() . "\n";
        }
    } else {
        // Command line args passed, just run the requested ones.
        foreach (array_slice($argv, 1) as $arg) {
            echo "#" . $arg . ". " . eval('return euler' . $arg . '();') . "\n";
        }
    }
}
?>